matkono/HealthSup
# Feature/refactoring authentication architecture

## What this PR does

- Refactor arquitecture.

## How to check if this PR is correct

- Features must be only in Authentication:
![image](https://user-images.githubusercontent.com/12301754/88466748-fabf7b00-cea5-11ea-8e4d-9ee7a4354659.png)

- Ensure AuthenticationAgent is functional
![image](https://user-images.githubusercontent.com/12301754/88466726-b6cc7600-cea5-11ea-881a-e4b2e21e3dfe.png)

- Ensure Authenticationuser is functional
![image](https://user-images.githubusercontent.com/12301754/88466729-bf24b100-cea5-11ea-8512-91d82f445e29.png)

- Ensure UpdatePassword is functional
![image](https://user-images.githubusercontent.com/12301754/88466732-c9df4600-cea5-11ea-9a4a-cb111c17b3b4.png)
